src/components/FollowButton.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { memo, useCallback, useEffect, useMemo, useState } from 'react'
+import { useSelector } from 'react-redux'
+import Follow from '@assets/images/follow.svg'
+import { BoxProps } from '@shopify/restyle'
+import { DebouncedTouchableOpacityBox } from './TouchableOpacityBox'
+import { useColors } from '../theme/themeHooks'
+import { useAppDispatch } from '../store/store'
+import { RootState } from '../store/rootReducer'
+import { followHotspot, unfollowHotspot } from '../store/hotspots/hotspotsSlice'
+import { Theme } from '../theme/theme'
+
+type Props = BoxProps<Theme> & {
+  address: string
+  colors?: { following: string; notFollowing: string }
+  handleChange?: (following: boolean) => void
+}
+const HotspotDetailsHandle = ({
+  address,
+  colors,
+  handleChange,
+  ...props
+}: Props) => {
+  const { grayPurple, followPurple } = useColors()
+  const dispatch = useAppDispatch()
+  const [following, setFollowing] = useState(false)
```

Comment:
is it necessary to have both local state and redux state for following?

Reply:
The round trip to the server takes a bit, that's why I have local state alongside redux state.